tree.rb

```diff
@@ -0,0 +1,22 @@
+
+
+def tree_crawler(directory)
+  # puts "entering tree_crawler('#{directory}')"
+  path = "#{directory}/"
+  Dir.foreach(directory) do |x|
+
+
+    if x == "." || x == ".."
```

Comment:
One pattern I like for this, `if ['.', '..'].include? x` works increasingly nicer for more exceptional cases.